tests/components/fivem/test_config_flow.py

```diff
@@ -0,0 +1,146 @@
+"""Test the FiveM config flow."""
+from unittest.mock import patch
+
+from fivem import FiveMServerOfflineError
+
+from homeassistant import config_entries
+from homeassistant.components.fivem.config_flow import DEFAULT_PORT
+from homeassistant.components.fivem.const import DOMAIN
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import RESULT_TYPE_CREATE_ENTRY, RESULT_TYPE_FORM
+
+USER_INPUT = {
+    CONF_NAME: "Dummy Server",
+    CONF_HOST: "fivem.dummyserver.com",
+    CONF_PORT: DEFAULT_PORT,
+}
+
+
+def __mock_fivem_info_success():
+    return {
+        "resources": [
+            "fivem",
+            "monitor",
+        ],
+        "server": "FXServer-dummy v0.0.0.DUMMY linux",
+        "vars": {
+            "gamename": "gta5",
+        },
+        "version": 123456789,
+    }
+
+
+def __mock_fivem_info_invalid():
+    return {
+        "plugins": [
+            "sample",
+        ],
+        "data": {
+            "gamename": "gta5",
+        },
+    }
+
+
+def __mock_fivem_info_invalid_gamename():
+    info = __mock_fivem_info_success()
+    info["vars"]["gamename"] = "redm"
+
+    return info
+
+
+async def test_show_config_form(hass: HomeAssistant) -> None:
+    """Test if initial configuration form is shown."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_form(hass: HomeAssistant) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] is None
+
+    with patch(
+        "fivem.fivem.FiveM.get_info_raw",
```

Comment:
Are we patching in the right place? It's usually safer to patch the name where we use it.

https://docs.python.org/3/library/unittest.mock.html#where-to-patch